Author:         Trent W. Buck <trentbuck@gmail.com>
Forwarded:      not-needed
Bug:            http://bugs.darcs.net/issue1508
Description:    Disable libHSdarcs.
 I have unilaterally decided to disable compilation and installation of
 libHSdarcs.  My reasons are as follows:
 .
  * no Debian package needs it;
  * the API is not stable; and
  * I do not with to switch back to CDBS (for hlibrary.mk).
 .
 If you need libHSdarcs, file a wishlist bug against Darcs explaining why, and
 in the meantime use "cabal install".
 .
 Not forwarded because upstream prefer to wait for a proper fix in Cabal 1.8.
Index: darcs-2.3.0/darcs.cabal
===================================================================
--- darcs-2.3.0.orig/darcs.cabal	2009-08-03 15:11:11.556565825 +1000
+++ darcs-2.3.0/darcs.cabal	2009-08-03 15:11:13.069071056 +1000
@@ -142,269 +142,6 @@
     cpp-options:    -DWIN32
     c-sources:      src/win32/send_email.c

--- ----------------------------------------------------------------------
--- darcs library
--- ----------------------------------------------------------------------
-
-Library
-  hs-source-dirs:   src
-  include-dirs:     src
-
-  exposed-modules:  CommandLine
-                    Crypt.SHA256
-                    Darcs.ArgumentDefaults
-                    Darcs.Arguments
-                    Darcs.Bug
-                    Darcs.CheckFileSystem
-                    Darcs.ColorPrinter
-                    Darcs.Commands
-                    Darcs.Commands.Add
-                    Darcs.Commands.AmendRecord
-                    Darcs.Commands.Annotate
-                    Darcs.Commands.Apply
-                    Darcs.CommandsAux
-                    Darcs.Commands.Changes
-                    Darcs.Commands.Check
-                    Darcs.Commands.Convert
-                    Darcs.Commands.Diff
-                    Darcs.Commands.Dist
-                    Darcs.Commands.Get
-                    Darcs.Commands.GZCRCs
-                    Darcs.Commands.Help
-                    Darcs.Commands.Init
-                    Darcs.Commands.MarkConflicts
-                    Darcs.Commands.Move
-                    Darcs.Commands.Optimize
-                    Darcs.Commands.Pull
-                    Darcs.Commands.Push
-                    Darcs.Commands.Put
-                    Darcs.Commands.Record
-                    Darcs.Commands.Remove
-                    Darcs.Commands.Repair
-                    Darcs.Commands.Replace
-                    Darcs.Commands.Revert
-                    Darcs.Commands.Rollback
-                    Darcs.Commands.Send
-                    Darcs.Commands.SetPref
-                    Darcs.Commands.Show
-                    Darcs.Commands.ShowAuthors
-                    Darcs.Commands.ShowBug
-                    Darcs.Commands.ShowContents
-                    Darcs.Commands.ShowFiles
-                    Darcs.Commands.ShowIndex
-                    Darcs.Commands.ShowRepo
-                    Darcs.Commands.ShowTags
-                    Darcs.Commands.Tag
-                    Darcs.Commands.TrackDown
-                    Darcs.Commands.TransferMode
-                    Darcs.Commands.Unrecord
-                    Darcs.Commands.Unrevert
-                    Darcs.Commands.WhatsNew
-                    Darcs.Compat
-                    Darcs.Diff
-                    Darcs.Email
-                    Darcs.External
-                    Darcs.FilePathMonad
-                    Darcs.Flags
-                    Darcs.Global
-                    Darcs.Gorsvet
-                    Darcs.Hopefully
-                    Darcs.IO
-                    Darcs.Lock
-                    Darcs.Match
-                    Darcs.Ordered
-                    Darcs.Patch
-                    Darcs.Patch.Apply
-                    Darcs.Patch.Bundle
-                    Darcs.Patch.Check
-                    Darcs.Patch.Choices
-                    Darcs.Patch.Commute
-                    Darcs.Patch.Core
-                    Darcs.Patch.Depends
-                    Darcs.Patch.FileName
-                    Darcs.Patch.Info
-                    Darcs.Patch.Match
-                    Darcs.Patch.MatchData
-                    Darcs.Patch.Non
-                    Darcs.Patch.Patchy
-                    Darcs.Patch.Permutations
-                    Darcs.Patch.Prim
-                    Darcs.Patch.Properties
-                    Darcs.Patch.Read
-                    Darcs.Patch.ReadMonads
-                    Darcs.Patch.Real
-                    Darcs.Patch.Set
-                    Darcs.Patch.Show
-                    Darcs.Patch.TouchesFiles
-                    Darcs.Patch.Viewing
-                    Darcs.Population
-                    Darcs.PopulationData
-                    Darcs.PrintPatch
-                    Darcs.ProgressPatches
-                    Darcs.RemoteApply
-                    Darcs.RepoPath
-                    Darcs.Repository
-                    Darcs.Repository.ApplyPatches
-                    Darcs.Repository.Cache
-                    Darcs.Repository.Checkpoint
-                    Darcs.Repository.DarcsRepo
-                    Darcs.Repository.Format
-                    Darcs.Repository.HashedIO
-                    Darcs.Repository.HashedRepo
-                    Darcs.Repository.Internal
-                    Darcs.Repository.InternalTypes
-                    Darcs.Repository.Motd
-                    Darcs.Repository.Prefs
-                    Darcs.Repository.Pristine
-                    Darcs.Repository.Repair
-                    Darcs.Resolution
-                    Darcs.RunCommand
-                    Darcs.Sealed
-                    Darcs.SelectChanges
-                    Darcs.Show
-                    Darcs.SignalHandler
-                    Darcs.SlurpDirectory
-                    Darcs.SlurpDirectory.Internal
-                    Darcs.Test
-                    Darcs.TheCommands
-                    Darcs.URL
-                    Darcs.Utils
-                    DateMatcher
-                    English
-                    Exec
-                    ByteStringUtils
-                    HTTP
-                    IsoDate
-                    Lcs
-                    OldDate
-                    Printer
-                    Progress
-                    RegChars
-                    SHA1
-                    Ssh
-                    URL
-                    UTF8
-                    Workaround
-
-  other-modules:    ThisVersion
-
-  c-sources:        src/atomic_create.c
-                    src/fpstring.c
-                    src/c_compat.c
-                    src/maybe_relink.c
-                    src/umask.c
-                    src/Crypt/sha2.c
-  cc-options:       -D_REENTRANT
-
-  if os(windows)
-    hs-source-dirs: src/win32
-    include-dirs:   src/win32
-    other-modules:  CtrlC
-                    System.Posix
-                    System.Posix.Files
-                    System.Posix.IO
-    cpp-options:    -DWIN32
-
-  if os(solaris)
-    cc-options:     -DHAVE_SIGINFO_H
-
-  build-depends:   base          < 4,
-                   regex-compat >= 0.71 && < 0.94,
-                   mtl          >= 1.0 && < 1.2,
-                   parsec       >= 2.0 && < 3.1,
-                   html         == 1.0.*,
-                   filepath     == 1.1.*,
-                   haskeline    >= 0.6.1 && < 0.7,
-                   hashed-storage >= 0.3.6 && < 0.4
-
-  if !os(windows)
-    build-depends: unix >= 1.0 && < 2.4
-
-  build-depends: base >= 3,
-                   old-time   == 1.0.*,
-                   directory  == 1.0.*,
-                   process    == 1.0.*,
-                   containers >= 0.1 && < 0.3,
-                   array      >= 0.1 && < 0.3,
-                   random     == 1.0.*
-
-  -- We need optimizations, regardless of what Hackage says
-  ghc-options:      -Wall -O2 -funbox-strict-fields
-  ghc-prof-options: -prof -auto-all
-
-  if flag(curl)
-    extra-libraries:   curl
-    includes:          curl/curl.h
-    cpp-options:       -DHAVE_CURL
-    c-sources:         src/hscurl.c
-    cc-options:        -DHAVE_CURL
-
-    if flag(curl-pipelining)
-      -- curl 7.19.1 has bug-free pipelining
-      if !os(windows)
-        pkgconfig-depends: libcurl >= 7.19.1
-      cpp-options:       -DCURL_PIPELINING -DCURL_PIPELINING_DEFAULT
-
-  if flag(http)
-      build-depends:    network == 2.2.*,
-                        HTTP    >= 3000.0 && < 4000.1
-      cpp-options:      -DHAVE_HTTP
-      x-have-http:
-
-  if !flag(curl) && !flag(http)
-      buildable: False
-
-  if flag(mmap) && !os(windows)
-    build-depends:    mmap >= 0.2
-    cpp-options:      -DHAVE_MMAP
-
-  if flag(bytestring)
-    build-depends:    bytestring >= 0.9.0 && < 0.10
-    cpp-options:      -DHAVE_BYTESTRING
-
-  if flag(zlib)
-    build-depends:    zlib >= 0.5.1.0 && < 0.6.0.0
-    cpp-options:      -DHAVE_HASKELL_ZLIB
-  else
-    extra-libraries:  z
-
-  if flag(utf8-string)
-    build-depends:    utf8-string == 0.3.*
-    cpp-options:      -DHAVE_UTF8STRING
-
-  -- The terminfo package cannot be built on Windows.
-  if flag(terminfo) && !os(windows)
-    build-depends:    terminfo == 0.3.*
-    cpp-options:      -DHAVE_TERMINFO
-
-  if flag(color)
-    x-use-color:
-
-  extensions:
-    CPP,
-    ForeignFunctionInterface,
-    BangPatterns,
-    PatternGuards,
-    MagicHash,
-    UndecidableInstances,
-    DeriveDataTypeable,
-    GADTs,
-    TypeOperators,
-    ExistentialQuantification,
-    FlexibleContexts,
-    FlexibleInstances,
-    ScopedTypeVariables,
-    KindSignatures,
-    TypeSynonymInstances,
-    Rank2Types,
-    RankNTypes,
-    GeneralizedNewtypeDeriving,
-    MultiParamTypeClasses
-
--- ----------------------------------------------------------------------
--- darcs itself
--- ----------------------------------------------------------------------
-
 Executable          darcs
   main-is:          darcs.hs
   hs-source-dirs:   src
@@ -526,112 +263,3 @@
     RankNTypes,
     GeneralizedNewtypeDeriving,
     MultiParamTypeClasses
-
--- ----------------------------------------------------------------------
--- unit test driver
--- ----------------------------------------------------------------------
-
-Executable          unit
-  main-is:          unit.lhs
-  hs-source-dirs:   src
-  include-dirs:     src
-  c-sources:        src/atomic_create.c
-                    src/fpstring.c
-                    src/c_compat.c
-                    src/maybe_relink.c
-                    src/umask.c
-                    src/Crypt/sha2.c
-
-  -- We need optimizations, regardless of what Hackage says
-  ghc-options:      -Wall -O2 -funbox-strict-fields -threaded
-  ghc-prof-options: -prof -auto-all
-
-  if !flag(test)
-    buildable: False
-  else
-    buildable: True
-    build-depends:   base          < 4,
-                     regex-compat >= 0.71 && < 0.94,
-                     mtl          >= 1.0 && < 1.2,
-                     parsec       >= 2.0 && < 3.1,
-                     html         == 1.0.*,
-                     filepath     == 1.1.*,
-                     QuickCheck   >= 2.1.0.0,
-                     HUnit        >= 1.0,
-                     test-framework             >= 0.2.2,
-                     test-framework-hunit       >= 0.2.2,
-                     test-framework-quickcheck2 >= 0.2.2
-
-
-  if !flag(zlib)
-    extra-libraries:  z
-
-  cc-options:       -D_REENTRANT
-
-  if os(windows)
-    hs-source-dirs: src/win32
-    include-dirs:   src/win32
-    other-modules:  CtrlC
-                    System.Posix
-                    System.Posix.Files
-                    System.Posix.IO
-    cpp-options:    -DWIN32
-    c-sources:      src/win32/send_email.c
-
-  if os(solaris)
-    cc-options:     -DHAVE_SIGINFO_H
-
-  if !os(windows)
-    build-depends: unix >= 1.0 && < 2.4
-
-  build-depends: base >= 3,
-                   old-time   == 1.0.*,
-                   directory  == 1.0.*,
-                   process    == 1.0.*,
-                   containers >= 0.1 && < 0.3,
-                   array      >= 0.1 && < 0.3,
-                   random     == 1.0.*
-
-  if flag(mmap) && !os(windows)
-    build-depends:    mmap >= 0.2
-    cpp-options:      -DHAVE_MMAP
-
-  if flag(bytestring)
-    build-depends:    bytestring >= 0.9.0 && < 0.10
-    cpp-options:      -DHAVE_BYTESTRING
-
-  if flag(zlib)
-    build-depends:    zlib >= 0.5.1.0 && < 0.6.0.0
-    cpp-options:      -DHAVE_HASKELL_ZLIB
-  else
-    extra-libraries:  z
-
-  -- The terminfo package cannot be built on Windows.
-  if flag(terminfo) && !os(windows)
-    build-depends:    terminfo == 0.3.*
-    cpp-options:      -DHAVE_TERMINFO
-
-  if flag(color)
-    x-use-color:
-
-  extensions:
-    CPP,
-    ForeignFunctionInterface,
-    BangPatterns,
-    PatternGuards,
-    MagicHash,
-    UndecidableInstances,
-    DeriveDataTypeable,
-    GADTs,
-    TypeOperators,
-    ExistentialQuantification,
-    FlexibleContexts,
-    FlexibleInstances,
-    ScopedTypeVariables,
-    KindSignatures,
-    TypeSynonymInstances,
-    Rank2Types,
-    RankNTypes,
-    GeneralizedNewtypeDeriving,
-    MultiParamTypeClasses
-    OverlappingInstances
